crowbar_framework/app/controllers/barclamp_controller.rb

```diff
@@ -271,103 +252,279 @@ def get_proposals_from_barclamps(barclamps)
-      render inline: {proposals: proposals, count: count, error: e.message}.to_json, cache: false
     end
   end
 
   #
-  # Provides the restful api call for
-  # Create Proposal Instance 	/crowbar/<barclamp-name>/<version>/proposals 	PUT 	Putting a json document will create a proposal
+  # Delete proposal
+  # Remove a specific proposal
+  # DELETE /crowbar/<barclamp-name>/<version>/proposals/<barclamp-instance-name>
   #
-  add_help(:proposal_create,[:name],[:put])
-  def proposal_create
-    Rails.logger.info "Proposal Create starting. Params #{params.inspect}"
-    controller = params[:controller]
-    orig_id = params[:name] || params[:id]
-    params[:id] = orig_id
-    answer = [500, "Server issue"]
-    begin
-      Rails.logger.info "asking for proposal of: #{params.inspect}"
-      answer = @service_object.proposal_create params
-      Rails.logger.info "proposal is: #{answer.inspect}"
-      unless answer[0] == 200
-        flash[:alert] = answer[1]
+  add_help(:proposal_delete, [:id], [:delete])
+  def proposal_delete
+    code, message = @service_object.proposal_delete(
+      params[:id]
+    )
+
+    raise Crowbar::Error::NotFound if code == 404
+    respond_to do |format|
+      case code
+      when 200
+        format.json do
+          head :ok
+        end
+      else
+        format.json do
+          render json: { error: message }, status: code
+        end
       end
-    rescue StandardError => e
-      flash_and_log_exception(e)
     end
+  end
+
+  #
+  # Commit proposal
+  # Commit a specific proposal to apply it
+  # POST /crowbar/<barclamp-name>/<version>/proposals/commit/<barclamp-instance-name>
+  #
+  add_help(:proposal_commit, [:id], [:post])
+  def proposal_commit
+    code, message = @service_object.proposal_commit(
+      params[:id]
+    )
+
+    raise Crowbar::Error::NotFound if code == 404
     respond_to do |format|
-      format.html {
-        return redirect_to barclamp_modules_path id: params[:controller] if answer[0] != 200
-        redirect_to proposal_barclamp_path controller: controller, id: orig_id
-      }
-      format.xml  {
-        return render text: flash[:alert], status: answer[0] if answer[0] != 200
-        render xml: answer[1]
-      }
-      format.json {
-        return render text: flash[:alert], status: answer[0] if answer[0] != 200
-        render json: answer[1]
-      }
+      case code
+      when 200
+        format.html do
+          redirect_to(
+            proposal_barclamp_path(
+              controller: params[:id],
+              id: params[:id]
+            )
+          )
+        end
+        format.json do
+          head :ok
+        end
+      when 202
+        format.html do
+          flash[:warning] = message
+
+          redirect_to(
+            proposal_barclamp_path(
+              controller: params[:id],
+              id: params[:id]
+            )
+          )
+        end
+        format.json do
+          head :ok
```

Comment:
Shouldn't we put the message here, instead of just ok? This happens when the proposal gets queued, and it's kind of nice to know even from json.